File: java/221.java
class Solution {
    public int maximalSquare(char[][] mat) {
        int res = 0, m = mat.length, n = mat[0].length;
        int dp[][] = new int[m + 1][n + 1];
        for (int i = 1; i <= m; i++) {
            for (int j = 1; j <= n; j++) {
                if (mat[i - 1][j - 1] == '1') {
                    dp[i][j] = Math.min(Math.min(dp[i - 1][j], dp[i - 1][j - 1]), dp[i][j - 1]) + 1;
                    res = (dp[i][j] > res) ? dp[i][j] : res;
                }
            }
        }
        return res * res;
    }
}

File: python3/221.py
class Solution:
    def maximalSquare(self, mat: List[List[str]]) -> int:
        def min(i, j, k):
            if i < j:
                return i if i < k else k
            return j if j < k else k

        res = 0
        m = len(mat)
        n = len(mat[0])
        dp = [[0] * (n + 1) for i in range(m + 1)]
        for i in range(1, m + 1):
            for j in range(1, n + 1):
                if mat[i - 1][j - 1] == "1":
                    dp[i][j] = min(dp[i - 1][j], dp[i - 1][j - 1], dp[i][j - 1]) + 1
                    res = dp[i][j] if (dp[i][j] > res) else res
        return res * res
